File: output/copilot/c/results/max-sum-of-rectangle-no-larger-than-k.c
/** https://leetcode.com/problems/max-sum-of-rectangle-no-larger-than-k/ */
//Given an m x n matrix matrix and an integer k, return the max sum of a rectangle in the matrix such that its sum is no larger than k.



int maxSumSubmatrix(int** matrix, int matrixSize, int* matrixColSize, int k){
    int i, j, m, n, sum, max = INT_MIN;
    for(i = 0; i < matrixSize; i++){
        for(j = i; j < matrixSize; j++){
            sum = 0;
            for(m = 0; m < matrixSize; m++){
                for(n = 0; n < matrixColSize[m]; n++){
                    sum += matrix[m][n];
                }
            }
            if(sum <= k){
                max = max > sum ? max : sum;
            }
        }
    }
    return max;
}
    

File: output/copilot/javascript/results/max-sum-of-rectangle-no-larger-than-k.js
/** https://leetcode.com/problems/max-sum-of-rectangle-no-larger-than-k/ */
//Given an m x n matrix matrix and an integer k, return the max sum of a rectangle in the matrix such that its sum is no larger than k.

/**
 * @param {number[][]} matrix
 * @param {number} k
 * @return {number}
 */
 var maxSumSubmatrix = function(matrix, k) {
    let max = Number.NEGATIVE_INFINITY;
        for (let i = 0; i < matrix.length; i++) {
        for (let j = 0; j < matrix[0].length; j++) {
            let sum = 0;
            for (let x = i; x < matrix.length; x++) {
                for (let y = j; y < matrix[0].length; y++) {
                    sum += matrix[x][y];
                    if (sum > max && sum <= k) {
                        max = sum;
                    }
                }
            }
        }
    }
    return max;
}
